src/libraries/System.Private.CoreLib/src/System/WASIp2/WasiEventLoop.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using WasiPollWorld.wit.imports.wasi.io.v0_2_0;
+
+namespace System
```

Comment:
I'm also thinking maybe this should be methods on `ThreadPool` class. I will try that on my PR.

Reply:
`ThreadPool` is a static class, and thus not accessible via `UnsafeAccessor`.  That's what I ran into when I tried to put them there.

BTW, the reason I named the folder WASIp2 was because I think this code will only be relevant for p2 and not for p3.

Reply:
@pavelsavara I've moved the `WASIp2` as you requested, but didn't rename it yet.  Wasn't sure if your 👍 was for leaving the functions in `Thread`, leaving the p2 on the name, or both.  Still happy to rename it though, if you prefer.

Reply:
I have it as `Wasi` after feedback about the case on my branch. I don't have p2 suffix as I believe we will have some generated and handwritten code there even in p3 and going forward. My thumb up meaning was : "ack" 😄

Reply:
I've renamed it to `Wasi`.

Reply:
thanks